### ISSUE TITLE
Very low performance

### ISSUE THREAD

**Issue by leonfancy** [ory/hydra#1477]
**Describe the bug**

I get very bad performance when I test Client Credentials Grant following the instrusction here: https://www.ory.sh/docs/performance/hydra. 

I setup Hydra in docker with resource limited to 2 CPU cores and 4G memory. And using memory database.

Then I send Client Credentials Grant traffic to Hydra with JMeter. I configured JMeter to send 100 requests per seconds. Below is my JMeter configuration:

[client_credentials.zip](https://github.com/ory/hydra/files/3305713/client_credentials.zip)

Here is the result:

![image](https://user-images.githubusercontent.com/4530475/59759575-fbf5a000-92c2-11e9-95e6-b28c10b45276.png)

The average latency is 3.67s, and the throught put is very low.

**Reproducing the bug**

1. Start Hydra:
```
docker run -d --name hydra --rm -p 4400:4444 -p 4401:4445 \
    --cpus="2" --memory="4g" \
    --network hydraguide \
    -e SECRETS_SYSTEM=abedfghijklmnopoqrsfdkfjdsa1234fdsafd \
    -e TTL_ACCESS_TOKEN=6h \
    -e URLS_CONSENT=http://127.0.0.1:8008/consent \
    -e URLS_LOGIN=http://127.0.0.1:8008/login \
    -e URLS_SELF_ISSUER=http://ncs.life:4400 \
    -e DSN=memory \
    -e WEBFINGER_JWKS_BROADCAST_KEYS=hydra.jwt.access-token \
    -e OAUTH2_HASHERS_BCRYPT_COST=8 \
   oryd/hydra:v1.0.0-rc.16 serve all --dangerous-force-http
```

2. Create a client:
```
docker run --rm --network hydraguide \
oryd/hydra:v1.0.0-rc.16 clients create \
--endpoint http://hydra:4445 \
--id SAMPLE_CLIENT \
--secret abcdefghijklmnopqrstuvwxyz0123456789 \
-g authorization_code,client_credentials \
--scope ncs \
-c http://127.0.0.1:8090/callback \
-r code \
--token-endpoint-auth-method client_secret_post
```

3. Run the JMeter test.

**Environment**

* Version: v1.0.0-rc.16
* Environment: Docker

**Comment by aeneasr** [comment]
Did you read https://www.ory.sh/docs/performance/hydra#bcrypt ?

**Comment by aeneasr** [comment]
I'm closing this because I think it's well documented. If you really need 100 concurrent clients using the OAuth 2.0 Client Credentials grant, make use of horizontal auto-scaling or scale the number of CPUs. Also, running these tests (presumably) on your local Desktop Docker environment isn't the best environment.

**Comment by leonfancy** [comment]
I did read that document. I see Hydra's throughput is 500 TPS in your report. So I create a same environment as yours (2 CPU, 4GB RAM), but I didn't get the same result. 

I didn't run the test on my own Desktop. Actually I first deploy it in K8s cluster, and find this issue. Then I create separate docker environment to run this test and get the result. I found only when I scale the CPU to 20 cores, hydra's thoughput can reach 100 TPS.
